packages/next/src/server/getSitemapProps.tsx

```diff
@@ -0,0 +1,62 @@
+import { GetServerSidePropsContext } from 'next';
+import { createPagesSitemap, createRootSitemapIndex, handleSitemapPath } from '../middleware/sitemaps/createSitemaps.js';
+import { HandleSitemapRequestsConfig } from '../middleware/sitemaps/handleSitemapRequests.js';
+
+export async function getSitemapProps(
+  ctx: GetServerSidePropsContext,
+  config: any,
+) {
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
+  const paramsIndex = ctx.req.url.indexOf('?');
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
+  const searchParamString = ctx.req.url.substr(paramsIndex);
+  const urlParams = new URLSearchParams(searchParamString);
+
+  if (!urlParams.get('sitemap')) {
+    // handle Root sitemap
+    // eslint-disable-next-line no-console
+    console.log('sitemap index');
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
+    const response = await createRootSitemapIndex(ctx.req, config, false);
+
+    ctx.res.setHeader('Content-Type', 'application/xml');
+
+    ctx.res.write(await response?.text());
+
+    ctx.res.end();
+  }
+
+  if (
+    urlParams.get('sitemap') &&
+    urlParams.get('sitemap') !== 'sitemap-faust-pages.xml'
+  ) {
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
```

Comment:
We might want to see how to make it typesafe:

https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/no-unsafe-argument.md